### DISCUSSION TITLE
Avoiding duplicated environments

### DISCUSSION THREAD

**Discussion by jimmywan** [tox-dev/tox#2193] [Q&A] [answered]
I have what seems like a simple use case where I have a few different tools (pylint, pytest, mypy, etc) and I want to run them in parallel, so I've defined separate testenvs for each, but I'd like to just share one set of dependencies so it doesn't have to eat up a whole bunch of disk.
Is there a tox-friendly way of doing this?

**Comment by gaborbernat** [comment, marked as answer]
There isn't at the moment. The problem is the more tools you install in the same environment the more likely you'll run into version conflicts, so you lose isolation. Also, pytest needs to be installed in the same environment your package is. 

What you describe would need to either:

- ability to merge configurations between environments
- ability to define tox environments with different targets

While both are possible, both are non-trivial to implement. So until someone volunteers to do it, it's unlikely to happen.

**Reply by jimmywan** [reply]
To be clear, version conflicts are a non issue. I don't really care/worry about version conflicts between tools. In a different project, I'm using poetry to detect/resolve those conflicts for me.

I'm using the same files to bootstrap my environment(s). Looks something like this:
```
[tox]
skipsdist=True

[testenv]
changedir = ./src/
deps =
	-r../requirements-dev.txt # they all get defined with the same set of dependencies

[testenv:prospector]
commands = prospector . --profile ../../linting/profile.yaml

[testenv:pylint]
commands = pylint --rcfile=../../linting/.pylintrc . ../tests

[testenv:bandit]
commands = bandit -r . -lll

[testenv:coverage]
commands =
	coverage run --omit '../.tox/*,../tests/*' --branch -m unittest discover -s ../tests
	coverage report -m
```

**Reply by gaborbernat** [reply]
Version conflicts are one of those everyone is quick to judge as non issue, until they run into.  And I'm not sure I'd recommend people to chdir to src but 🤷‍♂️
